### PR TITLE
llvm_6, llvm_7: build all default targets

### PR DESCRIPTION
**Pull request by orivej**
###### Motivation for this change

This makes LLVM tools (including dependent tools such as LLD) readily useful in
more situations, foresees such future additions as BPF and NVPTX, and brings
llvm_6 and newer on par with the current default llvm_5.

The list of targets was reduced in #52031, which was not necessary to fix cross compilation.
BPF target is needed for BCC in #53920.
`out` and `lib` size increases by 1/3, but the full LLVM is needed regardless, and this avoids having a partial LLVM.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by 7c6f434c — APPROVED** [review]

**Review by Mic92 — APPROVED** [review]

File: pkgs/development/compilers/llvm/7/llvm.nix [hunk 1]
```diff
@@ -28,9 +26,6 @@ let
   shortVersion = with stdenv.lib;
     concatStringsSep "." (take 1 (splitString "." release_version));
 
-  inherit
-    (import ../common.nix { inherit (stdenv) lib; })
-    llvmBackendList;
```
Comment (Mic92) [inline pkgs/development/compilers/llvm/7/llvm.nix]: Do we still need common.nix in that case?

Comment (orivej) [reply pkgs/development/compilers/llvm/7/llvm.nix]: No, indeed. Removed.

**Comment by dtzWill** [comment]
Assuming it evals, :+1: :).

**Comment by 7c6f434c** [comment]
Did aarch64 build succeed before?

**Comment by Mic92** [comment]
At least llvm_7 used to build on aarch64, because we are using this version for aarch64.

**Comment by orivej** [comment]
llvm_7 still builds on aarch64, but llvm_6 needs a backport of this fix: https://reviews.llvm.org/D50583. (This is similar to an older https://reviews.llvm.org/D28285.)

**Comment by orivej** [comment]
@GrahamcOfBorg build llvm_6 llvm_7

**Pull request merged**